synadm/api.py

```diff
@@ -73,26 +100,36 @@ def __init__(self, log, user, token, base_url, path, timeout, debug,
             HTTPConnection.debuglevel = 1
         self.verify = verify
 
-    def query(self, method, urlpart, params=None, data=None, token=None,
-              base_url_override=None, verify=None, *args, **kwargs):
+    def query(
+        self,
+        method: str,
+        urlpart: str,
+        *args: Any,
+        params: Optional[Dict[str, Any]] = None,
+        data: Optional[Dict[str, Any]] = None,
+        token: Optional[str] = None,
+        base_url_override: Optional[bool] = False,
+        verify: Optional[bool] = True,
+        **kwargs: Dict[str, Any],
+    ) -> Optional[Union[Dict[str, Any], List[Dict[str, Any]], None]]:
```

Comment:
This is pretty complicated and hard to read already. You wrote in the summary that we could actually receive a single "string" or number type without any wrapping dict or list? 

The current definition is telling that we get a list, a dict or None. Will this be accurate enough? Or should we even turn this into "returns anything or None"

Reply:
Looking at this again. The `, None` near the end of the line is redundant, actually `Optional` already tells that `None` could also be returned. This really tells what I intended to tell. Still same question: Making it more complicated would make sense yout thing or is it good enough?
```suggestion
Optional[Union[Dict[str, Any], List[Dict[str, Any]]]]
```